server/api/impl/aws/provision_app.go

```diff
@@ -45,21 +46,42 @@ type functionInstallData struct {
 //      |-- lambda_function.py
 //      |-- __pycache__
 //      |-- certifi/
-func (c *Client) ProvisionApp(releaseURL string) error {
+func (c *Client) ProvisionAppFromURL(releaseURL string, shouldUpdate bool) error {
 	zipFile, err := downloadFile(releaseURL)
 	if err != nil {
 		return errors.Wrapf(err, "can't install app from url %s", releaseURL)
 	}
-	zipReader, err := zip.NewReader(bytes.NewReader(zipFile), int64(len(zipFile)))
+
+	return c.ProvisionApp(zipFile, shouldUpdate)
+}
+
+func (c *Client) ProvisionAppFromFile(path string, shouldUpdate bool) error {
+	f, err := os.Open(path)
 	if err != nil {
-		return errors.Wrapf(err, "can't install app from url %s", releaseURL)
+		return errors.Wrapf(err, "can't read file from  path %s", path)
+	}
+
+	b, err := ioutil.ReadAll(f)
+	if err != nil {
+		return errors.Wrap(err, "can't read file")
+	}
+
+	return c.ProvisionApp(b, shouldUpdate)
+}
+
+func (c *Client) ProvisionApp(b []byte, shouldUpdate bool) error {
```

Comment:
I'm aware that it's bad practice to pass a byte slice, but it seemed fine to me given that the code will need a refactoring anyway.